cmd/cstor-volume-mgmt/controller/volume-controller/new_volume_controller.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volumecontroller
+
+import (
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/openebs/maya/cmd/cstor-volume-mgmt/controller/common"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	clientset "github.com/openebs/maya/pkg/client/clientset/versioned"
+	openebsScheme "github.com/openebs/maya/pkg/client/clientset/versioned/scheme"
+	informers "github.com/openebs/maya/pkg/client/informers/externalversions"
+)
+
+const volumeControllerName = "CStorVolume"
+
+// CStorVolumeController is the controller implementation for CStorVolume resources.
+type CStorVolumeController struct {
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+
+	// clientset is a openebs custom resource package generated for custom API group.
+	clientset clientset.Interface
+
+	// cStorVolumeSynced is used for caches sync to get populated
+	cStorVolumeSynced cache.InformerSynced
+
+	// deletedIndexer holds deleted resource to be retrived after workqueue
+	deletedIndexer cache.Indexer
```

Comment:
for handling with finalizer, deletedIndexer can be removed. Plz verify if it is being used elsewhere